python/faery/events_filter.py

```diff
@@ -223,73 +233,49 @@ def __iter__(self) -> collections.abc.Iterator[numpy.ndarray]:
 
 
 @typed_filter({"FiniteRegular"})
-class TimeSlice(events_stream.FiniteRegularEventsFilter):  # type: ignore
+class PacketSlice(events_stream.FiniteRegularEventsFilter):  # type: ignore
     def __init__(
         self,
         parent: stream.FiniteRegularStream[numpy.ndarray],
-        start: timestamp.Time,
-        end: timestamp.Time,
-        zero: bool,
+        start: int,
```

Comment:
Is there a reason you dropped the timestamp type?

Reply:
Yes, this is because this is part of the PacketSlice operator (selects packets by index), not the TimeSlice operator (selects packets in a given time range). The time slice operator still uses `timestamp.Time`. Non-regular streams have a `time_slice` function and regular streams have a `packet_slice` function.

I removed the TimeSlice operator for regular streams because the implementation had to make choices that were not transparent to the user. Because regular packets cover exactly `1e6 / frequency` µs, time slices on regular stream must either:
- downgrade the stream to a non-regular stream (which may not be what the user intended).
- raise an error if the difference between start and end is not an integer number of times `1e6 / frequency`. This can be frustrating if `1e6 / frequency` is not a round number (in which case the user must know how the library rounds such values).
- remove more or less than the time range provided by the user to preserve entire packets. Since time slices are likely to be used to synchronize streams, this could create very subtle and frustrating issues for users.